pydata_center/monitoring/tests/test_api.py

```diff
@@ -333,6 +387,28 @@ def test_missing_hostname_field(self):
             "'hostname' should be reported as missing"
         )
 
+    def test_receive_status_with_invalid_tags_format(self):
+        """
+        Test that a status update with an invalid format for 'tags'
+        (e.g., a string instead of a dict) returns a 400 Bad Request.
+        """
+        invalid_data = self._get_valid_status_data()
+        invalid_data['tags'] = 'This is not a valid json object'
+
+        response = self.client.post(self.url, data=invalid_data, format='json')
+
+        self.assertEqual(
+            response.status_code,
+            status.HTTP_400_BAD_REQUEST,
+            f'Expected 400 BAD REQUEST for invalid tags format, '
+            f'got {response.status_code}'
+        )
+        self.assertIn('tags', response.data)
+        self.assertIn(
+            'Invalid data. Expected a dictionary object',
+            str(response.data['tags'])
+        )
```

Comment:
Instead of using `str(...)` and checking for substring presence, assert more precisely to guard against accidental changes:
```
self.assertTrue(
    any('dictionary' in msg for msg in response.data['tags']),
    f"Expected a dictionary-related error message, got: {response.data['tags']}"
)
```